server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,94 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setWeights(String awarenessAttribute, Map<String, Double> weights, ActionListener<ClusterPutWeightedRoutingResponse> listener) {
```

Comment:
Is awarenessAttribute a key to the attribute? For eg. zone? If yes, then we need to reword this as attribute is key-value pair and not just key of the attribute